plugins/rbac/src/hooks/useRoles.ts

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { useAsync, useAsyncRetry, useInterval } from 'react-use';
+
+import { useApi } from '@backstage/core-plugin-api';
+import { usePermission } from '@backstage/plugin-permission-react';
+
+import {
+  policyEntityDeletePermission,
+  Role,
+  RoleBasedPolicy,
+} from '@janus-idp/backstage-plugin-rbac-common';
+
+import { rbacApiRef } from '../api/RBACBackendClient';
+import { RolesData } from '../types';
+import { getPermissions } from '../utils/rbac-utils';
+
+export const useRoles = (pollInterval?: number) => {
+  const rbacApi = useApi(rbacApiRef);
+  const {
+    loading: rolesLoading,
+    value: roles,
+    retry,
+  } = useAsyncRetry(async () => await rbacApi.getRoles());
+
+  const { loading: policiesLoading, value: policies } = useAsync(
```

Comment:
should we handle `error` case as well?

Reply:
I am showing "No roles found" when there's an error

![Screenshot 2023-11-28 at 1 30 10 PM](https://github.com/janus-idp/backstage-plugins/assets/22490998/d83b6f1d-ec0f-4093-ab75-c62c5411518a)